test/integration/controller/jobset_controller_test.go

```diff
@@ -219,32 +234,35 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 			gomega.Eventually(util.CheckNumJobs, timeout, interval).WithArguments(ctx, k8sClient, js).Should(gomega.Equal(util.NumExpectedJobs(js)))
 
 			// Perform a series of updates to jobset resources and check resulting jobset state after each update.
-			for _, update := range tc.updates {
-
-				// Fetch updated job objects so we always have the latest resource versions to perform mutations on.
-				var jobList batchv1.JobList
-				gomega.Expect(k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace))).Should(gomega.Succeed())
-
-				// Perform mutation on jobset if specified.
-				if update.jobUpdateFn != nil {
-					update.jobUpdateFn(&jobList)
+			for _, up := range tc.updates {
+
+				var jobSet jobset.JobSet
+				gomega.Eventually(k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &jobSet), timeout, interval).Should(gomega.Succeed())
+
+				if up.jobSetUpdateFn != nil {
+					up.jobSetUpdateFn(&jobSet)
+				} else if up.jobUpdateFn != nil {
+					// Fetch updated job objects so we always have the latest resource versions to perform mutations on.
+					var jobList batchv1.JobList
+					gomega.Eventually(k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)), timeout, interval).Should(gomega.Succeed())
```

Comment:
we need to check that the number of jobs is the expected one

Reply:
done.